### DISCUSSION TITLE
rauc extract is not extracting rauc bundle.

### DISCUSSION THREAD

**Discussion by samannazz** [rauc/rauc#1029] [Q&A]
Creating a Bundle (Host):

step 1: `mkdir content-dir/`

step 2: adding a custom rootfs.ext4  → {rauc/content-dir}, because `cp $SOURCE/rootfs.ext4 content-dir/` is not working.
 

step 3: creating manifest.raucm

```
cat >> content-dir/manifest.raucm << EOF
[update]
compatible=FooCorp Super BarBazzer
version=2019.01-1
[image.rootfs]
filename=rootfs.ext4
EOF
```

step 4: creating key and cert. {/rauc}

`openssl req -x509 -newkey rsa:4096 -nodes -keyout demo.key.pem -out demo.cert.pem -subj "/O=rauc Inc./CN=rauc-demo"
`
![Screenshot from 2022-12-27 15-08-46](https://user-images.githubusercontent.com/89451747/209658227-b98574d1-58b2-420d-8568-6087101c1545.png)

step 5: creating a bundle from this: {/rauc}

`rauc --cert demo.cert.pem --key demo.key.pem bundle content-dir/ update-2019.01-1.raucb
`
output:
![Screenshot from 2022-12-27 14-57-01](https://user-images.githubusercontent.com/89451747/209658349-7a528854-163e-4a32-b713-3db2e877df04.png)

The output of the command indicates that the RAUC bundle is being created in the "plain" format, which is a simple format that includes all of the files and directories in the bundle without any additional metadata or compression.

`rauc-Message: 13:59:54.136: No keyring given, skipping signature verification` indicates that the RAUC was unable to find a keyring to use for verifying the signature of the RAUC bundle.

How can i create keyring?

Step 6: now i am extracting ruac bundle so that i can launch virtual machine i.e. qemu.

![Screenshot from 2022-12-27 14-05-50](https://user-images.githubusercontent.com/89451747/209658553-066e718e-6e19-4290-b764-0ecd2e254379.png)

How can I solve this problem? Maybe it is because of the custom rootfs.ext4, or maybe I haven't created a keyring?

**Comment by ejoerns** [comment]
> How can I create keyring?

There are different ways for this, but since you have created a key pair, you should have a cert that can serve as a keyring.

> How can I solve this problem? Maybe it is because of the custom rootfs.ext4, or maybe I haven't created a keyring?

I have the feeling of having answered this or a similar question already on a different channel, but cannot find it quickly.

For reference: The error message you see comes from `unsquashfs` which attempts to create an output directory that already exist. This could be better handled in RAUC by checking first if the unpack directory exist and emitting a proper error from RAUC itself.

The solution is either way to remove the already-existing directory and to try again.
